tiledb/sm/consolidator/group_meta_consolidator.cc

```diff
@@ -76,58 +76,60 @@ Status GroupMetaConsolidator::consolidate(
   // Open group for reading
   auto group_uri = URI(group_name);
   Group group_for_reads(group_uri, storage_manager_);
-  RETURN_NOT_OK(group_for_reads.open(
-      QueryType::READ, config_.timestamp_start_, config_.timestamp_end_));
+  group_for_reads.open(
+      QueryType::READ, config_.timestamp_start_, config_.timestamp_end_);
 
   // Open group for writing
   Group group_for_writes(group_uri, storage_manager_);
-  RETURN_NOT_OK_ELSE(
-      group_for_writes.open(QueryType::WRITE),
-      throw_if_not_ok(group_for_reads.close()));
+  try {
+    group_for_writes.open(QueryType::WRITE);
+  } catch (const std::exception&) {
+    group_for_reads.close();
+    throw;
+  }
 
   // Swap the in-memory metadata between the two groups.
   // After that, the group for writes will store the (consolidated by
   // the way metadata loading works) metadata of the group for reads
-  Metadata* metadata_r;
-  auto st = group_for_reads.metadata(&metadata_r);
-  if (!st.ok()) {
-    throw_if_not_ok(group_for_reads.close());
-    throw_if_not_ok(group_for_writes.close());
-    return st;
-  }
-  Metadata* metadata_w;
-  st = group_for_writes.metadata(&metadata_w);
-  if (!st.ok()) {
-    throw_if_not_ok(group_for_reads.close());
-    throw_if_not_ok(group_for_writes.close());
-    return st;
+  Metadata *metadata_r, *metadata_w;
+  try {
+    metadata_r = group_for_reads.metadata();
+    metadata_w = group_for_writes.metadata();
+  } catch (const std::exception&) {
+    group_for_reads.close();
+    group_for_writes.close();
```

Comment:
It would be better to use two `try` blocks. The first catch would close only `group_for_reads`; the second would close both. This avoid potential problems where closing something that has never been opened is prohibited (as one might expect).